### ISSUE TITLE
Connecting to IPv6 address literals does not work

### ISSUE THREAD

**Issue by sesse** [gquintard/libvmod-gwist#5]
Hi,

I've tried creating backends using something like

`` set bereq.backend = gwist.backend("::1", 80, 60);``

and for whatever reason, I only get a “503 Backend fetch failed”. Using something that resolves to ::1, on the other hand, works fine.

**Comment by sesse** [comment]
On a closer look, it looks maybe as if something in bare_backend() tries to parse the IPv6 literal as something with a port in, yielding two different and somewhat incompatible meanings for the host field. Perhaps it shouldn't really do this if the port field is already populated? (Also, isn't “p + 3” for the port one character too far if the colon is at p[1]?)

**Comment by gquintard** [comment]
Hi @sesse,

I've trouble finding the section of code you are referring to, we pass the whole host to getaddrinfo, so we don't look for any colon whatsoever. Can you guide me on that one, what line and commit are you looking at?

**Comment by sesse** [comment]
Sure, it's at https://github.com/gquintard/libvmod-gwist/blob/master/src/vmod_gwist.c#L145

**Comment by gquintard** [comment]
You are right on both accounts (I was looking at an outdated local copy), that should be +2.

**Comment by gquintard** [comment]
Duh, sorry, this just hit me, the IPv6 must be enclosed in square-brackets...

**Comment by sesse** [comment]
Why? That makes no sense. The only reason to enclose an IPv6 address in square brackets is if you have a format where you can also specify a port, but the port is right there in the second parameter, not part of the string.

**Comment by sesse** [comment]
Well, then see my comment #2.

**Comment by gquintard** [comment]
88014143888fb2cd7e5d7a58228fdb84252ec103 ?

**Comment by sesse** [comment]
No, not the part about p + 3. “[…] yielding two different and somewhat incompatible meanings for the host field. Perhaps it shouldn't really do this if the port field is already populated?”

**Comment by gquintard** [comment]
It's actually modeled after URL format, and that's what a host header would use, so I don't think that's too strange.

Then there's the question about the behaviour of the port argument, I went for fallback, but you seem to prefer override. I'm honestly not sure one is way better than the one.
